### ISSUE TITLE
[API] Shows list sorted by name has wrong order

### ISSUE THREAD

**Issue by MGaetan89** [SiCKRAGETV/sickrage-issues#2646]
If you use the `shows ` API command and sort the shows by name, uppercase are sorted before lowercase. See "NCIS" and "Narcos" on the screenshot.

Shouldn't it be sorted independently of the case?

![screenshot_20150904-090111](https://cloud.githubusercontent.com/assets/1009664/9678269/889de95a-52e3-11e5-94b8-2671f10411ce.png)

**Comment by WebSpider** [comment]
Sorting is indeed missing from the CMD_Shows. We just fill the key differently, depending on the sorting method, and return an object.

This can be fixed in webapi.py:L2831 by adding a sort like shows.sort(key=show_name.lower)

**Comment by miigotu** [comment]
@WebSpider it's apparently already sorted somewhere, shouldnt the existing sort be fixed instead?

**Comment by MGaetan89** [comment]
I can fix it, but I have no idea where the sorting is done.
Are dictionary in python sorted automatically?

**Comment by miigotu** [comment]
Nope. Dicts are unordered. Is it sorted in the sql select? I havent looked for the code yet

**Comment by WebSpider** [comment]
Yeah, its the select that’s determining the order. We shouldn’t change the select, since it’s in reused functions all over the code, so post-sorting the object would be better.

 

Van: miigotu [mailto:notifications@github.com] 
Verzonden: vrijdag 9 oktober 2015 09:16
Aan: SiCKRAGETV/sickrage-issues <sickrage-issues@noreply.github.com>
CC: WebSpider <nils@familievogels.nl>
Onderwerp: Re: [sickrage-issues] Shows list sorted by name has wrong order (#2646)

 

Nope. Dicts are unordered. Is it sorted in the sql select? I havent looked for the code yet

**Comment by MGaetan89** [comment]
Ok. I will fix this by adding the sort in the API command, so it doesn't break anything else.
I may have some time this evening to fix this and #3204.

**Comment by MGaetan89** [comment]
I didn't managed to sort the list properly. If someone else want to give it a try, have fun.
